pkg/operation/botanist/component/nodelocaldns/nodelocaldns.go

```diff
@@ -0,0 +1,557 @@
+	prometheusScrape = true
+
+	nodeLocal     = common.NodeLocalIPVSAddress
+	domain        = gardencorev1beta1.DefaultDomain
+	serviceName   = "kube-dns-upstream"
+	configDataKey = "Corefile"
+)
+
+// Interface contains functions for a node-local-dns deployer.
+type Interface interface {
+	component.DeployWaiter
+	component.MonitoringComponent
+}
+
+// Values is a set of configuration values for the node-local-dns component.
+type Values struct {
+	// Image is the container image used for node-local-dns.
+	Image string
+	// VPAEnabled marks whether VerticalPodAutoscaler is enabled for the shoot.
+	VPAEnabled bool
+	// ForceTcpToClusterDNS enforces upgrade to tcp connections for communication between node local and cluster dns.
+	ForceTcpToClusterDNS bool
+	// ForceTcpToUpstreamDNS enforces upgrade to tcp connections for communication between node local and upstream dns.
+	ForceTcpToUpstreamDNS bool
+	// ClusterDNS is the ClusterIP of kube-system/coredns Service
+	ClusterDNS string
+	// DNSServer is the ClusterIP of kube-system/coredns Service
+	DNSServer string
+}
+
+// New creates a new instance of DeployWaiter for node-local-dns.
+func New(
+	client client.Client,
+	namespace string,
+	values Values,
+) Interface {
+	return &nodeLocalDNS{
+		client:    client,
+		namespace: namespace,
+		values:    values,
+	}
+}
+
+type nodeLocalDNS struct {
+	client    client.Client
+	namespace string
+	values    Values
+}
+
+func (c *nodeLocalDNS) Deploy(ctx context.Context) error {
+	data, err := c.computeResourcesData()
+	if err != nil {
+		return err
+	}
+	return managedresources.CreateForShoot(ctx, c.client, c.namespace, ManagedResourceName, false, data)
+}
+
+func (c *nodeLocalDNS) Destroy(ctx context.Context) error {
+	return managedresources.DeleteForShoot(ctx, c.client, c.namespace, ManagedResourceName)
+}
+
+// TimeoutWaitForManagedResource is the timeout used while waiting for the ManagedResources to become healthy
+// or deleted.
+var TimeoutWaitForManagedResource = 2 * time.Minute
+
+func (c *nodeLocalDNS) Wait(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResource)
+	defer cancel()
+
+	return managedresources.WaitUntilHealthy(timeoutCtx, c.client, c.namespace, ManagedResourceName)
+}
+
+func (c *nodeLocalDNS) WaitCleanup(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResource)
+	defer cancel()
+
+	return managedresources.WaitUntilDeleted(timeoutCtx, c.client, c.namespace, ManagedResourceName)
+}
+
+func (c *nodeLocalDNS) computeResourcesData() (map[string][]byte, error) {
+	var (
+		hostPathFileOrCreate = corev1.HostPathFileOrCreate
+		registry             = managedresources.NewRegistry(kubernetes.ShootScheme, kubernetes.ShootCodec, kubernetes.ShootSerializer)
+
+		serviceAccount = &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "node-local-dns",
+				Namespace: metav1.NamespaceSystem,
+			},
+		}
```

Comment:
See https://github.com/gardener/gardener/pull/5099/files#diff-0a8cd43ade12e817b6a3b8ec2f51716ebddd73e2d38bd794cfff8f7af3fef437R6, can you set the automount field to `false` here?